uint16 gpioId
uint32 pulseLengthInUs
---
bool hasSucceeded